controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteReaderGroupTask.java

```diff
@@ -0,0 +1,84 @@
+ */
+@Slf4j
+public class DeleteReaderGroupTask implements ReaderGroupTask<DeleteReaderGroupEvent> {
+
+    private final StreamMetadataStore streamMetadataStore;
+    private final StreamMetadataTasks streamMetadataTasks;
+    private final ScheduledExecutorService executor;
+
+    public DeleteReaderGroupTask(final StreamMetadataTasks streamMetaTasks,
+                                 final StreamMetadataStore streamMetaStore,
+                                 final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(streamMetaStore);
+        Preconditions.checkNotNull(streamMetaTasks);
+        Preconditions.checkNotNull(executor);
+        this.streamMetadataStore = streamMetaStore;
+        this.streamMetadataTasks = streamMetaTasks;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final DeleteReaderGroupEvent request) {
+      String scope = request.getScope();
+      String readerGroup = request.getRgName();
+      long requestId = request.getRequestId();
+      UUID readerGroupId = request.getReaderGroupId();
+      final RGOperationContext context = streamMetadataStore.createRGContext(scope, readerGroup);
+      return streamMetadataStore.getReaderGroupId(scope, readerGroup, context, executor)
+              .thenCompose(id -> {
+               if (!id.equals(readerGroupId)) {
+                      log.warn("Skipping processing of Reader Group delete request {} as UUIDs did not match.", requestId);
+                      return CompletableFuture.completedFuture(null);
+               }
+               return streamMetadataStore.getReaderGroupConfigRecord(scope, readerGroup, context, executor)
+                          .thenCompose(configRecord -> {
+                          if (!ReaderGroupConfig.StreamDataRetention.values()[configRecord.getObject().getRetentionTypeOrdinal()]
+                              .equals(ReaderGroupConfig.StreamDataRetention.NONE)) {
+                              String scopedRGName = NameUtils.getScopedReaderGroupName(scope, readerGroup);
+                              // update Stream metadata tables, if RG is a Subscriber
+                              Iterator<String> streamIter = configRecord.getObject().getStartingStreamCuts().keySet().iterator();
+                              return Futures.loop(() -> streamIter.hasNext(), () -> {
+                                     Stream stream = Stream.of(streamIter.next());
+                                     return streamMetadataStore.deleteSubscriber(stream.getScope(),
+                                            stream.getStreamName(), scopedRGName, null, executor);
+                                 }, executor);
+                              }
+                              return CompletableFuture.completedFuture(null);
+                          })
+                          .thenCompose(v -> streamMetadataTasks.deleteStream(scope,
```

Comment:
same comment as with create RG... again we can handle this in a separate PR.. or if we move RG creation to a separate event processor in future then it doesnt contest for the processing slots to be available (remember: each instantiation of concurrent event processor processes at most 1000 events concurrently per controller instance)..

Reply:
https://github.com/pravega/pravega/issues/5438